src/ansible/roles/keycloak/tasks/main.yml

```diff
@@ -0,0 +1,110 @@
+- name: Create Keycloak user/group
+  user:
+    name: keycloak
+    home: /opt/keycloak
+    system: yes
+    create_home: yes
+
+- name: Print keycloak_url
+  debug:
+    msg: "keycloak_url={{ keycloak_url }}"
+
+- name: Download software zipfile
+  get_url:
+    url: "{{ keycloak_url }}"
+    dest: /tmp
```

Comment:
You can use /tmp/keycloak.tgz or other file name, this will make the variables unneeded. Then the only fact you need is the keycloak_url, please remove it from facts and put it to this role.

Reply:
Changed.